src/serializer/modules.js

```diff
@@ -288,6 +289,8 @@ export class Modules {
       }
 
       return effects;
+    } catch (err) {
+      if (err instanceof FatalError) return undefined;
```

Comment:
Yes, evaluateNodeForEffects now rolls back fatal effects in its finally clause. See line 308 in realm.js.